File: app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/customcontrols/keyboard/LwjglCharSender.java
package net.kdt.pojavlaunch.customcontrols.keyboard;

import static net.kdt.pojavlaunch.platform.Platform.PLATFORM;

import android.view.KeyEvent;

import net.kdt.pojavlaunch.CallbackBridge;
import net.kdt.pojavlaunch.platform.Platform;

/** Sends keys via the CallBackBridge */
public class LwjglCharSender implements CharacterSenderStrategy {
    @Override
    public void sendBackspace() {
        CallbackBridge.sendKeyPress(KeyEvent.KEYCODE_DEL);
    }

    @Override
    public void sendEnter() {
        CallbackBridge.sendKeyPress(KeyEvent.KEYCODE_ENTER);
    }

    @Override
    public void sendChars(CharSequence chars) {
        PLATFORM.sendBulkUnicodeEvent(chars.toString(), CallbackBridge.getCurrentMods());
    }
}

File: app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/customcontrols/keyboard/LwjglCharSender.kt
package net.kdt.pojavlaunch.customcontrols.keyboard

import git.artdeell.dnbootstrap.glfw.GLFW
import net.kdt.pojavlaunch.CallbackBridge
import net.kdt.pojavlaunch.LwjglGlfwKeycode

class LwjglCharSender : CharacterSenderStrategy {
    override fun sendBackspace() {
        CallbackBridge.sendKeyPress(LwjglGlfwKeycode.GLFW_KEY_BACKSPACE)
    }

    override fun sendEnter() {
        CallbackBridge.sendKeyPress(LwjglGlfwKeycode.GLFW_KEY_ENTER)
    }

    override fun sendChars(chars: CharSequence) {
        GLFW.sendBulkUnicodeEvent(chars.toString(), CallbackBridge.getCurrentMods())
    }
}
